### PR TITLE
fix: Add example line to import template

### LINKED ISSUE #9874: Add an example line to the import spreadsheet template on the producers platform
This issue is part of #9872 , please see the parent issue for context and guidance on how to work on this issue.

We want to add 2 lines to the import template (generated by https://world.pro.openfoodfacts.org/cgi/generate_sample_import_file.pl ):

- a description line that should contain what we currently have in the comments in the header line (and then we can remove the comments)
- an example line

This issue is about adding the example line.

We also need add one first column that just contains "description" and "example" for those 2 lines, and nothing for the the other rows.

See the template proposal: https://docs.google.com/spreadsheets/d/148PWIsSfDRsj8gLJ7cHbQtdapIEscUwX/edit#gid=1221199414 :

We will need to change the importing so that rows that contain something in the first column are skipped.

Examples should be translated in different languages.
https://fr.pro.openfoodfacts.org/cgi/generate_sample_import_file.pl should have examples in French.

The examples translations are stored in a TSV file in GitHub: https://github.com/openfoodfacts/openfoodfacts-server/blob/main/conf/pro-platform/Import%20template%20-%20Example%20translations%20-%20Import%20sheet.tsv

The TSV file has the same columns as the English template, and one row per language, with the language code in the first column. That way it will be easy to edit the examples, add more languages etc. 

When generating the import template, we will need to:

- load the conf/pro-platform/Import%20template%20-%20Example%20translations%20-%20Import%20sheet.tsv TSV file
- select the line with language corresponding to the language of the import template we are generating ( $lc contains the language code). If there isn't a matching language (e.g. Italian is requested, but we only have English and French: use English).
- For each column of the generated import template, check if we have a corresponding column (same header name) in the example file. We cannot rely in the order of the columns, because it's possible that we will be adding / removing some columns. e.g. we will soon generate a "simple" version of the import template with less columns.

### PR DESCRIPTION
**Pull request by TheSussex**
### What

- I added format options for the example row
- I imported a .tsv file that contains example values for the import template columns and populated the columns with values based on the requested language.

- Fixes #9874

### CODE REVIEW

File: cgi/generate_sample_import_file.pl [hunk 1]
```diff
@@ -59,13 +59,35 @@
...
+chomp($header_line);
+my @headers = split("\t", $header_line);
+while (my $line = <$example_fh>) {
+    chomp($line);
+    my @values = split("\t", $line);
+    for my $i (0 .. $#headers) {
+        next if $headers[$i] eq 'lc';
+        $example_values_by_header{$headers[$i]} = $values[$i] if defined $values[$i];
+    }
+    last;
```
Comment (stephanegigandet) [inline cgi/generate_sample_import_file.pl:88]: Since you have "last;", you are only reading one line (the first one: the one for English)

Comment (stephanegigandet) [reply cgi/generate_sample_import_file.pl:88]: You should read all the lines, and populate a structure that contains example values for different languages. First you read the "lc" into $line_lc, and then you can create a structure $example_values_by_language_and_header{$line_lc} = {}, and you populate the headers in it.

**Comment by TheSussex** [comment]
I have updated the code as suggested but the behaviour is still not as expected. When I select french as the language, some fields like Ingredients and labels columns get populated with the french values but other columns remain empty even though there are french examples for them. 

@stephanegigandet

File: cgi/generate_sample_import_file.pl [hunk 2]
```diff
@@ -169,18 +191,8 @@
...
-			my $example_title = lang("example");
-
-			# Several examples?
-			if ($example =~ /,/) {
-				$example_title = lang("examples");
-			}
-			$comment .= $example_title . " " . $example . "\n\n";
-		}
+		my $line_lc = lc($request_ref->{lc} || 'en');
+    my $example_value = $example_values_by_language_and_header{$line_lc}{$field_ref->{text}} // '';
```
Comment (stephanegigandet) [inline cgi/generate_sample_import_file.pl:195]: The issue is that $field_ref->{text} can be in another language than English, but in the sample configuration file we have English.
The best solution is to add a row with field ids in the example configuration file. I made a PR based on yours here: https://github.com/openfoodfacts/openfoodfacts-server/pull/9985

**Comment by stephanegigandet** [comment]
> I have updated the code as suggested but the behaviour is still not as expected. When I select french as the language, some fields like Ingredients and labels columns get populated with the french values but other columns remain empty even though there are french examples for them.
> 
> @stephanegigandet

@TheSussex I think the best way to solve it is to add a row with field ids to the examples configuration spreadsheet:
https://github.com/openfoodfacts/openfoodfacts-server/pull/9985

**Comment by alexgarel** [comment]
@TheSussex thanks a lot for the PR, finally we merged https://github.com/openfoodfacts/openfoodfacts-server/pull/9985 that @stephanegigandet built upon your PR.

**Comment by TheSussex** [comment]
> @TheSussex thanks a lot for the PR, finally we merged #9985 that @stephanegigandet built upon your PR.

Thank you so much @stephanegigandet @alexgarel

I have gone through Stephane's fix and I understand now how the changes to the example spreadsheet helped solve the problem. Open source contributing is so educating and interesting!😊

**Pull request closed without merging**